app/controllers/api/files_controller.rb

```diff
@@ -0,0 +1,37 @@
+class API::FilesController < APIController
+  PUBLIC_FILES = Rails.public_path.join('files').children
+
+  def show
+    if file
+      send_file file, disposition: :inline
+    else
+      render json: { error: 'Fichier non trouvé' }, status: :not_found
+    end
+  end
+
+  def download
+    if file
+      send_file file, disposition: :attachment
+    else
+      render json: { error: 'Fichier non trouvé' }, status: :not_found
+    end
+  end
+
+  private
+
+  def file
+    PUBLIC_FILES.find do |public_file|
+      filename_from_path_without_extension(public_file) == file_params[:file_name]
+    end
```

Comment:
Clairement peu optimisé comme approche.

Pourquoi ne pas faire ça (un truc du genre) ?

```suggestion
    path = Rails.root.join('files', file_name)
    return unless File.exists?(path)
    path
```